lib/ansible/plugins/test/core.py

```diff
@@ -168,6 +168,9 @@ def version_compare(value, version, operator='eq', strict=None, version_type=Non
     if strict is not None and version_type is not None:
         raise errors.AnsibleFilterError("Cannot specify both 'strict' and 'version_type'")
 
+    if not value:
```

Comment:
I think we need to check both `version` and `value`.  Otherwise just catch the `AttributeError` in the below try/except and give this error there.

Reply:
Both should be checked.

Do we want one error message, or a separate error message for each combination (value, version, both)?

Reply:
Catching the `AttributeError` was the first approach I took. Then I switched to failing early by checking up front. I didn't think about checking both `value` and `version`, though.

I'll just test both here and adjust the error message.

Reply:
I went with separate error messages for both. The language was too ambiguous trying to cram it into one message.

Reply:
I'm open to feedback on the language. It's tough trying to communicate the _input_ version value versus the version comparison value. They're both version comparison values. 🤯